### ISSUE TITLE
Detect stuck queues

### ISSUE THREAD

**Issue by bergie** [the-grid/guv#62]
Guv could be useful for detecting, and potentially remedying queues that are deadlocked and not moving

**Comment by jonnor** [comment]
guv does not know the identity of messages nor their contents (nor should it).
So to monitor whether jobs are being processed in an orderly fashion, we'd have to look at the ACK ratio - which I think RabbitMQ provides. And then compare it to the theoretical throughput (which we have an estimate for). If this is way off, say 5-10 times too low, we could then ask Heroku to restart the role/dyno-type.
There are however, no guarantees that this will clear the problem. But it may in some cases.

This is not a condition which should occur, so it should also be reported as an error.

**Comment by bergie** [comment]
Yeah, I think just detecting it and reporting an error would be a good start here
